File: init/content/labs/01/TraceDemo.java
public class TraceDemo {
    private static int sum(int n) {
        if (n <= 0) {
            return 0;
        }
        return n + sum(n - 1);
    }
    public static void main(String[] args) {
        int[] arr = new int[5];
        for (int i = 0; i < arr.length; i++) {
            int n = (int)Math.pow(2, i);
            arr[i] = n;
        }

        int total = sum(4);

        String str = "hello";
        for (int i = 0; i < str.length(); i++) {
            System.out.println(str.charAt(i));
        }
    }
}

File: init/content/labs/01/trace_demo.py
def sum(n: int):
    if n <= 0:
        return 0
    return n + sum(n - 1)


nums = [x for x in range(7)]
for i in range(len(nums)):
    nums[i] = pow(2, i)

total = sum(4)
print("total", total)

s = "hello"
for c in s:
    print(c)
